### PR TITLE
Remove legacy authentication code.

### LINKED ISSUE #7281: Remove legacy authentication code.
_FEATURE OVERVIEW_
-------------------
#### User Story
As a developer on the DoSomething.org team, I don't want a lot of legacy code sticking around and confusing me when I'm trying to work on login/registration/signup code. So let's get rid of it.

#### Additional Information (optional)
🔥 🔥 🔥 

#### Why This Matters
burn it down

#### Current Workaround w/o Feature (if applicable)
🍂

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request removes code for legacy login & registration, and a lot of the user hooks for syncing Phoenix changes into Northstar. This simplifies the codebase by getting rid of unused code & a ton of if-statements, and also will allow devs to login via Northstar QA when working locally. _Nice._

#### How should this be reviewed?
To test, pull down this branch & run `ds pull stage --db` to get a fresh copy of the staging database.

⚠️ __You will see a ton of errors until you pull the staging database (after checking out this branch), since the Northstar module is now required and we all had it disabled.__ I'll make another announcement in #phoenix once this gets merged.

#### Any background context you want to provide?
🔥

#### Relevant tickets
Fixes #7281

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love
- [x] Post a message in #phoenix if this includes something that causes a rebuild!

### CODE REVIEW

**Comment by DFurnes** [comment]
Re-created this PR from #7282 because of a weird StyleCI commit hook error.

**Comment by DFurnes** [comment]
I'm going to do some more testing on my local, but this should be ready for review! I'd probably recommend going commit-by-commit since the diff is pretty hefty otherwise.

**Comment by DFurnes** [comment]
I've removed the old methods for syncing data into Northstar, since otherwise it'll be easy to shoot ourselves in the foot and corrupt QA data from local instances. This should not affect anything in normal day-to-day use, since they're only referenced from old scripts.

File: lib/modules/dosomething/dosomething_user/dosomething_user.module [hunk 1]
```diff
@@ -310,7 +310,7 @@ function dosomething_user_form_alter(&$form, $form_state, $form_id) {
 
       $form['#after_build'][] = 'dosomething_user_remove_extra_values_from_address_field';
       if ($validate_address) {
-        $form['#after_build'][] = 'dosomething_user_add_validation_attributes_to_address_fields';
+//        $form['#after_build'][] = 'dosomething_user_add_validation_attributes_to_address_fields';
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_user/dosomething_user.module:313]: Commented out (and above)? Not sure if maybe it gets removed in later commit, but adding a comment here.

File: lib/themes/dosomething/paraneue_dosomething/includes/form.inc [hunk 2]
```diff
@@ -205,33 +205,23 @@ function paraneue_dosomething_form_dosomething_payment_form_alter(&$form, &$form
 function paraneue_dosomething_form_user_profile_form_alter(&$form, &$form_state, $form_id) {
   $is_resetting_password = $form['account']['current_pass']['#access'] === false;
 
-  // If user was created by Northstar & isn't resetting their password, redirect them to edit profile there.
-  if (module_exists('dosomething_northstar') && dosomething_user_get_field('field_created_by_openid_connect') && !$is_resetting_password) {
+  // Unless the user is resetting their password, send them to Northstar's profile form.
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/includes/form.inc:208]: Oh, so we don't want them going to profile.dosomething.org when resetting passwords? Cool 😄

**Review by weerd — APPROVED** [review]
Sat down with a nice cup of ☕️  and 🕵️ 

Looks good! Added a couple comments but nothing to stop merging.

File: lib/modules/dosomething/dosomething_user/dosomething_user.module (hunk 1 above)
Comment (DFurnes) [reply lib/modules/dosomething/dosomething_user/dosomething_user.module:313]: Oh, yeah this should probably be removed (and we should start doing that in Northstar).

File: lib/themes/dosomething/paraneue_dosomething/includes/form.inc (hunk 2 above)
Comment (DFurnes) [reply lib/themes/dosomething/paraneue_dosomething/includes/form.inc:208]: For now! Hopefully we're able to launch Northstar's password reset flow sometime next sprint and we can remove this as well.

**Comment by DFurnes** [comment]
🙇! Will address your comment & do some further cleanup in another PR.

**Pull request merged**